servers/src/grin/sync.rs

```diff
@@ -167,12 +167,30 @@ pub fn run_sync(
 											"fast_sync: error = {:?}. restart fast sync",
 											sync_error
 										);
-										si.fast_sync_reset();
-										sync_error_need_clear = true;
+										sync_need_restart = true;
 									}
 									drop(clone);
 								}
-								if sync_error_need_clear {
+
+								// check peer connection status of this sync
+								if let Some(ref peer) = si.fast_sync_peer {
```

Comment:
Perhaps extract those 3 cascading `if`s as a utility function? `sync.rs` is starting to balloon a bit and this main loop even more so.

Reply:
Agree 😺    Actually I plan to do a refactor on the whole `run_sync()`, which is quite hard to do a code review because of a lot of line wrapped.

but prefer to do it in next PR to avoid mixing up with this fix.